### PR TITLE
upload-to-box: don't create a new thread every second

### PR DESCRIPTION
**Pull request by kpengboy**
Is there some reason we were doing the old behavior?

### CODE REVIEW

**Comment by kpengboy** [comment]
@jvperrin ^

**Comment by jvperrin** [comment]
Before there was a problem that if it slept at all it would upload very slowly, because it would spend most of its time sleeping and not actually sending data. That's why it's using the timer instead of sleeping.

This change should work, since the progress-printing thread is separate from the data uploading one, but I'd need to try it and see (unless you have already). I don't think it should use a timer though, because if it does, then it will still start a new printing thread every second, which is not desirable. Just calling the `print_progress` function in the new thread should be sufficient.

File: modules/ocf_backups/files/upload-to-box [hunk 1]
```diff
@@ -353,7 +353,9 @@ def main(args):
     # Clear the screen and start printing the progress bar every second
     if not args.quiet:
         subprocess.call(('clear',))
-        print_progress(multi, stats, 0)
+        progress_thread = threading.Timer(1, print_progress, (multi, stats))
```
Comment (jvperrin) [inline modules/ocf_backups/files/upload-to-box]: This is the line I meant, I think this should just be a regular thread, not a timer now.

Comment (kpengboy) [inline modules/ocf_backups/files/upload-to-box]: Timers are merely threads that start after an initial delay.

Comment (jvperrin) [inline modules/ocf_backups/files/upload-to-box]: Ok, but then why have the delay? It's not like it does anything here.

Comment (kpengboy) [inline modules/ocf_backups/files/upload-to-box]: Fair point.

**Pull request merged**